examples/taskruns/taskrun.yaml

```diff
@@ -66,6 +66,9 @@ spec:
     - --dockerfile=${inputs.params.pathToDockerFile}
     - --destination=${outputs.resources.builtImage.url}
     - --context=${inputs.params.pathToContext}
+    env:
```

Comment:
:thinking: interesting, this *shouldn't* be required as `$HOME` should be `/builder/home` and… it works on the CI :sweat_smile: cc @bobcatfish because I might overlook something :upside_down_face:

Reply:
The issue suffered me several hours, then I found this: 
[https://github.com/GoogleContainerTools/kaniko/issues/290](url)

Reply:
hm it might have something to do with what the default account in the namespace has permission to do (for example, mine always has permission to push to registries within the same project)

I think it makes sense that by default kaniko wouldn't be using anything in `/builder/home/.docker/`

For context @vincent-pli, were you providing a service account to this TaskRun when running it, or just running it with the default service account in the namespace?

Reply:
@bobcatfish 
I provide specific service account rather that default one, which include two secret, one for git another for docker.

By default, the kaniko try to find docker configuration file at /kaniko/.docker/config.json(the path define by env parameter: DOCKER_CONFIG), it's empty. 
The "creds-init" just generate the file to /builder/home/.docker/

That's why I suggest to rewrite DOCKER_CONFIG in the task definition.